Dev

NikolasGeorgaklis opened the pull request:
created .gitignore and started ignoring .DS_Store

NikolasGeorgaklis committed:
Start ignoring .DS_Store

NikolasGeorgaklis committed:
Create .gitignore

NikolasGeorgaklis committed:
Added Camera Permissions

As of iOS 10, the following camera and photo library usage descriptions are required in the Info.plist to access the camera roll as well as the camera to take photos.

NikolasGeorgaklis committed:
Start ignoring .UserInterfaceState.xcuserstate
git rm --cached NikoBracket.xcworkspace/xcuserdata/nikog.xcuserdatad/UserInterfaceState.xcuserstate

NikolasGeorgaklis committed:
Update .gitignore

NikolasGeorgaklis committed:
Added cocoa touch classes for the profile and editProfile view controllers and removed comments from appdelegate.m

For each view controller, I have to create a cocoa touch class to program the behavior of the view controller. I also removed unnecessary comments from appdelegate.m.

NikolasGeorgaklis committed:
Add ParseUI pod

Add parseUI dependency to simplify logging in/signing up PFUser's and displaying lists of PFObject's.

NikolasGeorgaklis committed:
Add elements to ProfileViewController and EditProfileViewController

I added the text boxes and pfimageview to the profile to display the user information and profile picture. I also added text fields and the pfimageview to the editviewcontroller to create the UI through which the user will edit their profile.

vaguilar commented on NikoBracket/ProfileViewController.m:
```diff
@@ -18,7 +20,10 @@ @implementation ProfileViewController
+    PFUser *user = [PFUser currentUser];
     
+//    NSData *imageData = [[NSData alloc] initWithContentsOfURL:[NSURL URLWithString:]];
+//    self.pfp.image = [UIImage imageWithData:imageData];
```
more commented code

vaguilar commented on NikoBracket/ProfileViewController.m:
```diff
@@ -22,7 +23,9 @@ - (void)viewDidLoad {
-//    NSData *imageData = [[NSData alloc] initWithContentsOfURL:[NSURL URLWithString:]];
+    
+    
+//    NSData *imageData = [[NSData alloc] initWithContentsOfURL:[NSURL URLWithString:user.]];
```
looks like this was committed by accident? might be good to just remove it.

vaguilar commented on NikoBracket/ProfileViewController.m:
```diff
@@ -0,0 +1,35 @@
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+}
+*/
```
if the commented code isn't necessary than we can omit it from the commit

vaguilar commented:
re: committing the `Pods/` directory. more of a comment, I've never done this before but apparently its ok according to the cocoa pods docs https://guides.cocoapods.org/using/using-cocoapods.html#should-i-check-the-pods-directory-into-source-control TIL

NikolasGeorgaklis committed:
Remove uncommented code and display default profile picture in ProfileViewController and EditProfileViewController

NikolasGeorgaklis committed:
Implement edit profile photo feature

A user can now use the camera or camera roll to select a photo to display as their profile picture. Once selected, the backend updates with the most recently selected photo.

vaguilar commented on NikoBracket/ProfileViewController.m:
```diff
@@ -23,21 +23,10 @@ - (void)viewDidLoad {
-    // Get the new view controller using [segue destinationViewController].
-    // Pass the selected object to the new view controller.
 }
-*/
```
usually for these fixes its best to go back and try to fix the commit in place, or at least separate it out to its own commit. theres a lot of commits already in this stack so this may be difficult now, but we should do this next time.
